exp/mutators/azureasomanagedcontrolplane.go

```diff
@@ -80,6 +81,10 @@ func SetManagedClusterDefaults(ctrlClient client.Client, asoManagedControlPlane
 			return err
 		}
 
+		if err := setManagedClusterCredentials(ctx, cluster, managedClusterPath, managedCluster); err != nil {
```

Comment:
`asoctl import` doesn't populate any configuration for ASO to generate a kubeconfig Secret for the cluster, so CAPZ will now take responsibility for doing that if it's not already defined. This smooths out adoption by letting users essentially pipe the `asoctl` output straight to `kubectl apply`.